GoogleChromeLabs/ps-analysis-tool
# Feature: Classify cookies by its use case and origin

## Description



- Adds script to download [open-cookie-databse](https://raw.githubusercontent.com/rtCamp/Open-Cookie-Database/add-wordpress-cookies/open-cookie-database.csv) and formats it as a JSON file ( called cookieDB ).
- Adds category tags on the basis of 
  - Origin - Third party/First Party.
  - Use case - Functional/Preferences/Analytics/Marketing/Unknown.
- Adds interception for cookies sent in responses.

## Relevant Technical Choices

- Update `cookieData` type to support new `analytics` value.
- Update webpack config to bundle cookie DB file.
- Update manifest file to add cookie DB file as an extension asset.
- Update `ListItem` component to display origin and use case classification tags.  
- 4 Utility functions added.
  - `isFirstParty`- util function for origin based classification.
  - `parseRequestCookieHeader` -  for parsing request cookies.
  - `fetchCookieDictionary` - for fetching cookieDB.
  - `findAnalyticsMatch` - for finding a cookie's analytics in the cookieDB.
- Add `npm run cookie-db:update` command to fetch, format and create a JSON file (CookieDB) 



## Testing Instructions



- Uninstall any previous versions of the extension if installed.
- Follow instructions in the README.MD to build and load the extension.
- Open a new tab and go to any website 
- Open the devtools panel and go to "Privacy sandbox" tab.
- A scrollable list of cookies should be visible that were used in the network requests generated by the tab.

## Additional Information:



## Screenshot/Screencast
![Screenshot 2023-06-23 at 10 09 45 AM](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/53055971/32cb7841-e08c-4b7f-96b7-2e9f32cbfd69)



---



Fixes #3
